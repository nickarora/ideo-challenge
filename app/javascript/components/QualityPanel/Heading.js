import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const Container = styled.header`
  color: white;
  border-radius: 0;
  font-size: 24px;
  padding: 10px 15px;
`

const Heading = ({ quality }) => (
  <Container
    style={{
      background: quality.color,
    }}
  >
    {quality.name}
  </Container>
)

Heading.propTypes = {
  quality: PropTypes.shape({
    name: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
  }).isRequired,
}

export default Heading
