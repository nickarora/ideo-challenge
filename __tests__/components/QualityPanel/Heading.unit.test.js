import Heading from '~/components/QualityPanel/Heading'

describe('Heading', () => {
  it('renders the quality name with the provided background color', () => {
    const wrapper = shallow(
      <Heading
        quality={{
          name: 'Foo',
          color: '#333',
        }}
      >
        CONTENT
      </Heading>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
