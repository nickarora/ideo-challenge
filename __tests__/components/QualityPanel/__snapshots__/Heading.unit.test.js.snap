// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders the quality name with the provided background color 1`] = `
<styled.header
  style={
    Object {
      "background": "#333",
    }
  }
>
  Foo
</styled.header>
`;
